projects/step-core/src/lib/components/predefined-options-input/predefined-options-input.component.ts

```diff
@@ -0,0 +1,63 @@
+import { KeyValue } from '@angular/common';
+import { Component, Input, Optional, TrackByFunction } from '@angular/core';
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+
+type OnChange = (value: string) => void;
+type OnTouch = () => void;
+
+@Component({
+  selector: 'step-predefined-options-input',
+  templateUrl: './predefined-options-input.component.html',
+  styleUrls: ['./predefined-options-input.component.scss'],
+})
+export class PredefinedOptionsInputComponent implements ControlValueAccessor {
+  @Input() label?: string;
+  @Input() placeholder?: string = '';
+  @Input() predefinedOptions!: KeyValue<string, string>[];
+  @Input() predefinedOptionsLabel!: string;
+  @Input() showRequiredMarker?: boolean;
+  @Input() isParentInvalid?: boolean;
+  @Input() isParentTouched?: boolean;
```

Comment:
the idea is that if we have a component that implements ```ControlValueAccessor``` we should be able to use it within other such components, these 2 properties make it possible for it to be controlled from the outside when necessary; imo, we should account for that in all ```ControlValueAccessor``` components so that they are composable